include/oneapi/dpl/pstl/utils.h

```diff
@@ -128,6 +128,21 @@ class __pstl_assign
     }
 };
 
+template <typename _Comp, typename _Proj>
+struct __compare
+{
+    //'mutable' is to relax the requirements for a user comparator or/and projection type operator() may be non-const
+    mutable _Comp __comp;
+    mutable _Proj __proj;
+
+    template <typename _Xp, typename _Yp>
+    bool
+    operator()(const _Xp& __x, const _Yp& __y) const
+    {
+        return __comp(__x, __y);
```

Comment:
Should we call the projection operator on `__x` and `__y` before passing to the comparator?

Reply:
sure.. Good catch, thank you, Adam!